extensions/cba_settings/trim.h

```diff
@@ -0,0 +1,33 @@
+#include <string>
+
+using namespace std;
+
+const string WHITESPACE = " \n\r\f\t\v";
+
+string trim_left(string str) {
+    size_t first = str.find_first_not_of(WHITESPACE);
+
+    if (first == string::npos)
+        return "";
+    else
+        return str.substr(first);
+}
+
+string trim_right(string str) {
+    size_t last = str.find_last_not_of(WHITESPACE);
+
+    if (last == string::npos)
+        return "";
+    else
+        return str.substr(0, last + 1);
+}
+
+string trim(string str) {
```

Comment:
Example for a more efficient trim that doesn't copy the string
https://github.com/dedmen/armake/blob/cpp/src/utils.cpp#L240